data-processor/build.gradle

```diff
@@ -9,6 +9,7 @@ dependencies {
 	compileOnly mn.micronaut.core.processor
 
 	testAnnotationProcessor mn.micronaut.inject.java
```

Comment:
Added this so we can add some entities used for tests as java classes vs always building them in runtime. And didn't want to add all to data-tck since those are creating tables in all tests and we don't need it for some tests, like this PR.